src/app/clusters/occupancy-sensor-server/occupancy-sensor-server.cpp

```diff
@@ -16,14 +16,144 @@
+
+    if (index < ArraySize(mHoldTimeLimitsStructs))
+    {
+        endpointIndex = index;
+        return CHIP_NO_ERROR;
+    }
+    return CHIP_ERROR_NOT_FOUND;
+}
+
+HoldTimeLimitsManager HoldTimeLimitsManager::mInstance;
+
+HoldTimeLimitsManager & HoldTimeLimitsManager::Instance()
+{
+    return mInstance;
+}
+
+CHIP_ERROR HoldTimeLimitsManager::Init()
+{
+    // Prevents re-initializing
+    VerifyOrReturnError(!mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+
+    for (size_t i = 0; i <= kOccupancySensingServerMaxEndpointCount; i++)
+    {
+        if (emberAfContainsServer(EndpointId(i), OccupancySensing::Id))
+        {
+            Structs::HoldTimeLimitsStruct::Type holdTimeLimitsInit;
+
+            // Set up some sane initial values for hold time limits structures
+            holdTimeLimitsInit.holdTimeMin     = 1;
+            holdTimeLimitsInit.holdTimeMax     = 300;
+            holdTimeLimitsInit.holdTimeDefault = 10;
+
+            HoldTimeLimitsManager::Instance().SetHoldTimeLimitsStruct(EndpointId(i), holdTimeLimitsInit);
+        }
+    }
+
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+
+    mIsInitialized = true;
+    return CHIP_NO_ERROR;
+}
+
+// AttributeAccessInterface
+CHIP_ERROR HoldTimeLimitsManager::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::HoldTimeLimits::Id: {
+
+        Structs::HoldTimeLimitsStruct::Type * holdTimeLimitsStruct =
+            HoldTimeLimitsManager::Instance().mHoldTimeLimits.GetHoldTimeLimitsStruct(aPath.mEndpointId);
+        Structs::HoldTimeLimitsStruct::Type res;
+        res.holdTimeMin     = holdTimeLimitsStruct->holdTimeMin;
+        res.holdTimeMax     = holdTimeLimitsStruct->holdTimeMax;
+        res.holdTimeDefault = holdTimeLimitsStruct->holdTimeDefault;
+        return aEncoder.Encode(res);
+    }
+    default:
+        return CHIP_NO_ERROR;
+    }
+}
+
+Structs::HoldTimeLimitsStruct::Type * HoldTimeLimitsManager::GetHoldTimeLimitsStruct(EndpointId endpoint)
+{
+    Structs::HoldTimeLimitsStruct::Type * holdTimeLimitsStruct = mHoldTimeLimits.GetHoldTimeLimitsStruct(endpoint);
+    return holdTimeLimitsStruct;
+}
+
+CHIP_ERROR HoldTimeLimitsManager::SetHoldTimeLimitsStruct(EndpointId endpoint,
+                                                          Structs::HoldTimeLimitsStruct::Type & holdTimeLimitsStruct)
+{
+    ReturnErrorOnFailure(mHoldTimeLimits.SetHoldTimeLimitsStruct(endpoint, holdTimeLimitsStruct));
+    return CHIP_NO_ERROR;
+}
+
+} // namespace OccupancySensing
+} // namespace Clusters
+} // namespace app
+} // namespace chip
 
 using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::OccupancySensing;
 
+void chip::app::Clusters::OccupancySensing::emberAfPluginOccupancyClusterServerPostInitCallback(EndpointId endpoint) {}
```

Comment:
Why is this here?  I know this is pre-existing, but it's just dead code that should be removed, no?

Reply:
Yes, you're right, dead code removed.